File: preppi_slim/SCR/setup_genome.pl
#!/usr/bin/perl

use strict;
use warnings;

use MODS::Globals;
use MODS::Genome;
use MODS::Pipeline;
# use MODS::Methods;
use Getopt::Std;

my $PGM = $0;                   #name of program
$PGM =~ s#.*/##;     
my $usage = <<USAGE;
USAGE:
        $PGM genome_name -f sequences.fa [-d] [-q] 
        Parameters:
		
		genome_name: Name of the genome. 
		-f sequences.fa: Fasta file with the sequences of the target proteins
		-d Toggle on debug mode.
                -q Set up the Pipeline, but don't qsub.             
USAGE

if(@ARGV<1 ) {
  print STDERR $usage ;
  exit(-1);
  }

my $gname=shift;

my %opts;
getopts('dqf:m:',\%opts);
my $debug=$opts{d} ? 'yes' : 'no';
my $execute=$opts{q} ? 'no' : 'yes' ;

die("You must specify a fasta file containing at least one sequence") if not defined $opts{f};

my $genome=new MODS::Genome(gname => $gname );
$genome->init($opts{f}) if defined $opts{f};

my $pipeline=new MODS::Pipeline(name=>"Setup",gname=>$gname,debug=>$debug);
unlink $pipeline->{stepsfn}, "$pipeline->{stepsfn}.focus";
$pipeline->add_step("IUPRED");

exit(0) if defined $opts{q};
$pipeline->qsub();

File: preppi_slim/SCR/setup_genome.py
#!/usr/bin/env python3
import argparse
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from MODS.Genome import Genome
from MODS.Pipeline import Pipeline


def main():
    parser = argparse.ArgumentParser(description="Create a SLiM genome and run IUPred")
    parser.add_argument("genome")
    parser.add_argument("-f", "--fasta", required=True)
    parser.add_argument("-d", "--debug", action="store_true")
    parser.add_argument("-q", "--no-submit", action="store_true")
    args = parser.parse_args()
    Genome(gname=args.genome).init(args.fasta)
    pipeline = Pipeline(name="Setup", gname=args.genome, debug="yes" if args.debug else "no")
    Path(pipeline.stepsfn).unlink(missing_ok=True)
    Path(pipeline.stepsfn + ".focus").unlink(missing_ok=True)
    pipeline.add_step("IUPRED")
    pipeline.qsub("no" if args.no_submit else "yes")


if __name__ == "__main__":
    main()
